### DISCUSSION TITLE
fix it to work on your React Native project

### DISCUSSION THREAD

**Discussion by 0xGaryWu** [0xPolygon/matic.js#229] [General]
last week, I import 'maticnetwork/matic.js' to my React Native Project( which is an Android/iOS mobile Wallet App), I encountered 2 issues.
1. Error: Unable to resolve module `tty` from `node_modules/@metamask/controllers/node_modules/@maticnetwork/maticjs/dist/matic.node.js`, occurred at app starting;
2. Dynamic require defined at line8; not supported by Metro, occurred when you call MaticPOSClient functions;

to fix it, you can try to dependent  'patch-package', and put my [2patches](https://github.com/GoPocketStudio/MISC/tree/main/polygon) into your project's patches folder.

**Comment by 0xGaryWu** [comment]
it's Not very elegant

**Comment by rawatnaresh** [comment]
Hello @liangwu, is it working for you in latest version?

**Reply by ujjwalgupta94** [reply]
those issues are not part of v3, i mean those has been fixed in v3. V3 now uses static server for getting those abi. 

so it should work now for latest version.

**Reply by rawatnaresh** [reply]
Hi @ujjwalgupta94 , Thank you for your response but I'm still getting the import error in `react-native`. I've created an issue here.
https://github.com/maticnetwork/matic.js/issues/336
